### PR TITLE
Add error handling section to docs

### PR DESCRIPTION
**Pull request by werwty**
closes #2760
https://pulp.plan.io/issues/2760

### CODE REVIEW

**Comment by werwty** [comment]
ok test

**Comment by goosemania** [comment]
ok test

File: docs/contributing/style_guide.rst [hunk 1]
```diff
@@ -65,3 +65,40 @@ Encoding
...
+.. _error-handling:
+
+Error Handling
+--------------
+
+Errors in Tasks
+***************
+
+All uncaught exceptions in a task are treated as fatal exceptions. The task is then marked as
+failed. The error traceback, description, and code are returned to the user under the 'error'
```
Comment (goosemania) [inline docs/contributing/style_guide.rst]: Is comma needed here? s/description,/description/ ?

File: docs/contributing/style_guide.rst [hunk 2]
```diff
@@ -65,3 +65,40 @@ Encoding
...
+attribute of the task object.
```
Comment (goosemania) [inline docs/contributing/style_guide.rst]: Suggestion: refer to attributes and classes, prefix with ~ to have an object name and not the full path.
What do you think? Something like this:
"the :attr:\`~pulpcore.app.models.Task.error\` attribute of the :class:\`~pulpcore.app.models.Task\` object"

File: docs/plugins/plugin-writer/basics.rst [hunk 3]
```diff
@@ -64,3 +64,10 @@ plugin components, particularly for both models and viewsets.
 Subclassing Importer, Publisher
 -------------------------------
 
+Error Handling
+--------------
+
+Please see the :ref:`error-handling` section in the code guidelines.
+
+Non fatal exceptions should be recorded with :meth:`pulpcore.plugin.tasking.Task.append_non_fatal_error`
+These non-fatal exceptions will be returned as a non_fatal_errors attribute on the returned task.
```
Comment (goosemania) [inline docs/plugins/plugin-writer/basics.rst]: Same suggestion here for the attribute
"returned as a :attr:\`~pulpcore.app.models.Task.non_fatal_errors\` attribute"

**Pull request merged**